# Myntra Product Data Analysis

## Business Objective

Analyze Myntra's product dataset to identify pricing trends, brand performance, discount strategies, customer preferences, and product popularity.

The goal is to generate business insights that help improve pricing, marketing strategies, and customer satisfaction.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


## Load Dataset

In [2]:
df = pd.read_csv("../data/myntra_dataset_ByScraping.csv")

## Display First Five Rows


In [3]:
df.head()

,brand_name,pants_description,price,MRP,discount_percent,ratings,number_of_ratings
0,WROGN,Men Loose Fit Cotton Jeans,1374.0,2499.0,0.45,4.2,57.0
1,Flying Machine,Men Slim Fit Jeans,1829.0,2999.0,0.39,4.6,5.0
2,Roadster,Men Pure Cotton Jeans,974.0,2499.0,0.61,3.6,1100.0
3,Bene Kleed,Relaxed Fit Denim Jeans,873.0,2299.0,0.62,4.0,4800.0
4,Levis,Men 511 Slim Fit Jeans,1478.0,2899.0,0.49,4.3,264.0


## Dataset Shape

In [4]:
df.shape

(52120, 7)

### Observation

The dataset contains 52,120 products and 7 features.

## Column Names

In [5]:
df.columns

Index(['brand_name', 'pants_description', 'price', 'MRP', 'discount_percent',
       'ratings', 'number_of_ratings'],
      dtype='str')

### Observation

The dataset contains product details, pricing, discounts, ratings, and brand information.

## Dataset Information

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 52120 entries, 0 to 52119
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   brand_name         52120 non-null  str    
 1   pants_description  52120 non-null  str    
 2   price              52120 non-null  float64
 3   MRP                52120 non-null  float64
 4   discount_percent   52120 non-null  float64
 5   ratings            52120 non-null  float64
 6   number_of_ratings  52120 non-null  float64
dtypes: float64(5), str(2)
memory usage: 4.5 MB


### Observation

The dataset contains both numerical and categorical variables.

Next step is to investigate missing values and clean the data.

## Summary Statistics

In [7]:
df.describe()

,price,MRP,discount_percent,ratings,number_of_ratings
count,52120.000000,52120.000000,52120.000000,52120.000000,52120.000000
mean,1594.515445,3180.398438,1.648256,3.997794,91.568937
std,1495.972325,2201.883218,4.687529,0.420404,433.918513
min,337.000000,499.000000,0.020000,1.000000,5.000000
25%,989.000000,2499.000000,0.400000,3.800000,16.000000
50%,1439.000000,2999.000000,0.500000,4.000000,35.000000
75%,1829.000000,3499.000000,0.630000,4.200000,74.000000
max,54000.000000,72000.000000,64.000000,5.000000,30700.000000


## Random Sample

In [8]:
df.sample(5)

,brand_name,pants_description,price,MRP,discount_percent,ratings,number_of_ratings
2063,Levis,Men 511 Slim Fit Jeans,1478.0,2899.0,0.49,4.3,264.0
40257,Integriti,Men Slim Fit Mid-Rise Jeans,1006.0,3049.0,0.67,4.2,18.0
31735,Mast & Harbour,Men Smart Tapered Fit Jeans,839.0,2399.0,0.65,3.8,53.0
139,Flying Machine,Men Relaxed Fit Jeans,1402.0,2299.0,0.39,3.9,117.0
48099,SPYKAR,Men Relaxed Fit Jeans,1924.0,3499.0,0.45,4.4,79.0


# Initial Findings

Based on the initial exploration:

- The dataset loaded successfully.
- The dataset contains more than 52,000 products.
- No missing values were observed.
- The dataset includes pricing, discounts, customer ratings, and brand information.
- Data types appear suitable for analysis.
- The dataset is ready for a detailed data quality assessment.